iapetus/porinMembraneSystem.py

```diff
@@ -0,0 +1,160 @@
+
+    .. _ParMed: https://parmed.github.io/ParmEd/html/index.html
+    .. _Structure: https://parmed.github.io/ParmEd/html/api/parmed/parmed.structure.html?highlight=structure#parmed.structure.Structure
+    .. _minimizeEnergy: http://docs.openmm.org/latest/userguide/application.html?highlight=modeller#energy-minimization
+
+    """
+
+    def __init__(self, ligand_name, system, topology, positions, platform, membrane=None):
+
+        """
+        Parameters
+        ----------
+        ligand_name : str
+            The name of the ligand
+        system : object
+            An OpenMM system
+        topology : object
+            An OpenMM topology
+        positions : array
+            Positions stored in an OpenMM context
+        platform : object
+            Platform used by OpenMM
+        membrane : str
+            The name of the membrane
+
+
+        Attributes
+        ----------
+        ligand : str
+            The residue name assigned to the ligand in the AMBER parameter/topology file (.prmtop)
+        structure : object
+            ParMed Structure_ of the minimized ligand-porin-membrane system
+
+        .. _Structure : .. _Structure: https://parmed.github.io/ParmEd/html/api/parmed/parmed.structure.html?highlight=structure#parmed.structure.Structure
+
+        """
+
+        # Retrieve the residue name of the ligand from the AMBER parameter/topology
+        # file (.prmtop)
+        with open(ligand_name + '.prmtop') as f:
+            res_label = False
+            for line in f:
+                if (res_label):
+                    self.ligand = str(next(f).split()[0])
+                    break
+                if line.startswith('%'+'FLAG RESIDUE_LABEL'):
+                    res_label = True
+
+        structure =  pmd.openmm.load_topology(topology,
+                                              system=system,
+                                              xyz=positions)
+        ligand_structure = pmd.load_file(ligand_name + '.prmtop', xyz=ligand_name + '.inpcrd')
+        # Save porin indices
+        top = md.Topology.from_openmm(topology)
+        if membrane is not None:
+            porin_indices = topo.select('(protein and not resname ' + membrane + ')')
+        else:
+            porin_indices = topo.select('protein')
+        # Place the ligand at the geometrical center of the porin's bottom and get
+        # the ParMed Structure
+        self.structure = self._PlaceLigand(structure, ligand_structure, porin_indices)
+        # Select atoms to be freezed during the minimization
+        top = md.Topology.from_openmm(self.structure.topology)
+        atoms_to_freeze = new_topo.select('protein or resname  ' + membrane + ' or resname ' + self.ligand)
+        # Perform the minimization of the ligand-porin-membrane
+        self._MinimizeEnergy(atoms_to_freeze, platform)
+
+    def _PlaceLigand(self, structure, ligand_structure, porin_indices):
```

Comment:
We follow the [PEP8 convention](https://www.python.org/dev/peps/pep-0008/) regarding method naming: This should be `_place_ligand`.

See our notes about naming in our [software development guide](https://github.com/choderalab/software-development) in the [`PYTHON_CODING` section](https://github.com/choderalab/software-development/blob/master/PYTHON_CODING.md).